packages/core/admin/ee/admin/content-manager/pages/EditView/InformationBox/InformationBoxEE.js

```diff
@@ -19,30 +20,77 @@ export function InformationBoxEE() {
     initialData,
     isCreatingEntry,
     layout: { uid },
+    isSingleType,
   } = useCMEditViewDataManager();
+  const [formattedError, setFormattedError] = useState(null);
   const { put } = useFetchClient();
   const activeWorkflowStage = initialData?.[ATTRIBUTE_NAME] ?? null;
+  const hasReviewWorkflowsEnabled = Object.prototype.hasOwnProperty.call(
+    initialData,
+    ATTRIBUTE_NAME
+  );
   const { formatMessage } = useIntl();
   const { formatAPIError } = useAPIErrorHandler();
+  const toggleNotification = useNotification();
 
-  const { workflows: { data: [workflow] = [] } = {} } = useReviewWorkflows();
+  const { workflows: { data: workflows, isLoading: workflowIsLoading } = {} } =
+    useReviewWorkflows();
+  // TODO: this works only as long as we support one workflow
+  const workflow = workflows?.[0] ?? null;
 
-  const { error, isLoading, mutateAsync } = useMutation(async ({ entityId, stageId, uid }) => {
-    const {
-      data: { data },
-    } = await put(`/admin/content-manager/collection-types/${uid}/${entityId}/stage`, {
-      data: { id: stageId },
-    });
+  const { error, isLoading, mutateAsync } = useMutation(
+    async ({ entityId, stageId, uid }) => {
+      const typeSlug = isSingleType ? 'single-types' : 'collection-types';
+
+      const {
+        data: { data },
+      } = await put(`/admin/content-manager/${typeSlug}/${uid}/${entityId}/stage`, {
+        data: { id: stageId },
+      });
+
+      return data;
+    },
+    {
+      onSuccess() {
+        toggleNotification({
+          type: 'success',
+          message: { id: 'notification.success.saved', defaultMessage: 'Saved' },
+        });
+      },
+    }
+  );
 
-    return data;
-  });
+  // if entities are created e.g. through lifecycle methods
+  // they may not have a stage assigned. Updating the entity won't
+  // set the default stage either which may lead to entities that
+  // do not have a stage assigned for a while. By displaying an
+  // error by default we are trying to nudge content-editors into
+  // assigning a stage.
+  useEffect(() => {
+    if (activeWorkflowStage === null && !workflowIsLoading && !isCreatingEntry) {
+      setFormattedError(
+        formatMessage({
+          id: 'content-manager.reviewWorkflows.stage.select.placeholder',
+          defaultMessage: 'Select a stage',
+        })
+      );
+    }
+  }, [activeWorkflowStage, workflowIsLoading, isCreatingEntry, formatMessage]);
 
-  // stages are empty while the workflow is loading
-  const options = (workflow?.stages ?? []).map(({ id, name }) => ({ value: id, label: name }));
-  const formattedError = error ? formatAPIError(error) : null;
+  useEffect(() => {
+    if (error) {
+      setFormattedError(formatAPIError(error));
+    }
+  }, [error, formatAPIError]);
```

Comment:
Does all this have to be in effects? Can't it be derived? I'm not sure I see the value, when you update the select the things save and the errors would be removed no?

Reply:
You are right, and I had a feeling you would say that. I mostly split it for readability ...

The error is not reset when changing the value of the stage, because right now I am not resetting `initialData` to the new entity which is returned by the stage change API call. This might also be the reason for the bug Jamie uncovered. I'd like to address this in a follow-up PR and left the bit out for now, so that we can see the bug for both (collections and single-types).

Reply:
Okay lets address it in another PR, id normally ask for a TODO comment but i trust you 😛